### ISSUE TITLE
better error message for more than max columns

### ISSUE THREAD

**Issue by camerondavison** [uniVocity/univocity-parsers#247]
currently it just throws a random ArrayIndexOutOfBounds exception
```
! java.lang.ArrayIndexOutOfBoundsException: 512
! at com.univocity.parsers.common.ParserOutput.valueParsed(ParserOutput.java:316)
! at com.univocity.parsers.csv.CsvParser.parseRecord(CsvParser.java:155)
! at com.univocity.parsers.common.AbstractParser.parse(AbstractParser.java:123)
```
It would be nice if it threw something that said "header contains more than max columns".

**Comment by jbax** [comment]
Are you using a recent version? You should be getting a message from `AbstractParser#303`:

```
message = "Length of parsed input (" + length + ") exceeds the maximum number of characters defined in" +
						" your parser settings (" + settings.getMaxCharsPerColumn() + "). "

```

**Comment by camerondavison** [comment]
I am using `com.univocity:univocity-parsers:2.6.3`

**Comment by camerondavison** [comment]
I wrote an example. I did see the error message that you mentioned above when I changed the parser method to
```java
final CsvParser parser = new CsvParser(s);
parser.beginParsing(newReader("/issues/github_247/input-100.txt"));
parser.getContext().parsedHeaders();
```

**Comment by jbax** [comment]
Thanks a lot for reporting this bug and sending through the pull request. It's very helpful.

I've fixed the bug and released a 2.6.4-SNAPSHOT build that includes the adjustment. The final version will be released in about a week.

Thank you for using our parsers!
